### ISSUE TITLE
Enhancement: [prefer-nullish-coalescing] Ignore env vars

### ISSUE THREAD

**Issue by koooge** [typescript-eslint/typescript-eslint#9162]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-nullish-coalescing/

### Description

Hi there,
A proposal of prefer-nullish-coalescing, it would be useful to have an option to ignore env vars i.e. `process.env` also import.meta.env. Env vars could be given like `FOO= node foo.js`, and the FOO is a string `''` in node.js. However, in many case it expects the same as `unset FOO`, so `??` does not work as expected since `''` is valid string.

### Fail

```TypeScript
// `$ FOO= node foo.js`
const foo = process.env.FOO || 'foobar'; // should be ignored
```

### Pass

```TypeScript
// `$ FOO= node foo.js`
const foo = process.env.FOO || 'foobar'; // prefer-nullish-coalescing suggests `??`, but `foobar` is the expected
```

**Comment by bradzacher** [comment]
The rule has no concept of platform or framework specific APIs. We purposely ***do not*** encode any specifics into any of our rules.

In this instance we would not be encoding special semantics around node's env var API.

It's also not uncommon that you would want the nullish coalescing behaviour instead of the logical or. For example if you want a default value but you want someone to be able to set any other value.

Theres no hard and fast rule that works here.

Which is why there's an option to ignore specific types of primitives based on your preferences.
https://typescript-eslint.io/rules/prefer-nullish-coalescing/#ignoreprimitives
